cpp/CMakeLists.txt

```diff
@@ -740,15 +740,38 @@ add_library(cudf::cudf ALIAS cudf)
 
 if(CUDF_BUILD_TESTUTIL)
   add_library(
```

Comment:
Should we only be building this separate library if `CUDF_BUILD_STREAMS_TEST_UTIL`?

Reply:
We need to always build this library as the functions it defines are used by methods in `cudftestutil`

Reply:
Great. @vyasr clarified a related question here, I didn't realize cudftestutil was part of libcudf (I thought it was only part of our tests). https://github.com/rapidsai/cudf/pull/12956#discussion_r1139215640

Reply:
Indeed, this part is somewhat confusing. The test util is shipped as part of libcudf so that other packages can use our test utilities without needing to load our test binaries.